### PR TITLE
Add option to calculate higher-order moments in velocity

### PR DESCRIPTION
**Pull request by rosteen**
This currently works for moment 1, but I need to add handling for higher orders since those will end up as `m2/s2` (for order 2) and so on. Also still need to add tests.

### CODE REVIEW

**Comment by rosteen** [comment]
Tests added, docs updated, good to go (well, ready for review at least).

File: jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py [hunk 1]
```diff
@@ -62,6 +69,11 @@ def __init__(self, *args, **kwargs):
 
         self.moment = None
 
+        self.output_unit = SelectPluginComponent(self,
```
Comment (kecnry) [inline jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:72]: do we want this added to the user API?

File: jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py [hunk 2]
```diff
@@ -126,6 +148,14 @@ def calculate_moment(self, add_data=True):
         if data_wcs:
             data_wcs = data_wcs.swapaxes(0, 1)  # We also transpose WCS to match.
         self.moment = CCDData(analysis.moment(slab, order=n_moment).T, wcs=data_wcs)
+        if self.output_unit == "Velocity":
```
Comment (kecnry) [inline jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:151]:
```suggestion
        if n_moment > 0 and self.output_unit == "Velocity":
```

File: jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.vue [hunk 3]
```diff
@@ -35,6 +35,38 @@
...
+        v-model.number="reference_wavelength"
+        hint="Rest wavelength of the line of interest"
+        persistent-hint
+        :rules="[() => reference_wavelength !== '' || 'This field is required',
+                 () => reference_wavelength >=0 || 'Wavelength must be positive']"
+        ></v-text-field>
+      </v-col>
+      <v-col cols="3" justify="left">{{dataset_spectral_unit}}</v-col>
+    </v-row>
+    </v-div>
```
Comment (kecnry) [inline jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.vue:68]: I have some style suggestions, but they get too messy as a review suggestion, so I'll submit a PR to yours and feel free to decide what you want to keep.  That addresses:
* replaces nonexisting `v-div` with `div`
* removes extra padding around reference wavelength
* moves the unit from a column to a suffix on the text field (I'm torn on this one, we use both patterns in jdaviz though right now, so pick whichever you'd prefer.  But if you'd rather columns, add `class="row-no-outside-padding"` to the `v-row`, use `dataset_spectral_unit.replace('Angstrom', '&#8491;')` and will need to fix the vertical alignment of the unit)
* removes unused vertical space for no hint on the radio buttons
* stacks radio buttons horizontally in a row (to be consistent with links control and to save a little space)
* fixes indentation

**Review by kecnry — COMMENTED** [review]
Seems to work well!  Just a few small suggestions and then I can take a final look!

**Comment by kecnry** [comment]
> This currently works for moment 1, but I need to add handling for higher orders since those will end up as m2/s2 (for order 2) and so on

Is this still the case?

**Comment by rosteen** [comment]
> > This currently works for moment 1, but I need to add handling for higher orders since those will end up as m2/s2 (for order 2) and so on
> 
> Is this still the case?

No, the code works for arbitrary `n_moment` now.

File: jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py (hunk 1 above)
Comment (rosteen) [reply jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:72]: Good call, I can do that after tagup.

File: jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py (hunk 2 above)
Comment (rosteen) [reply jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py:151]: Ah right, I didn't actually add any handling to switch back to `Wavelength` if you go back to moment 0. Good catch.

**Review by kecnry — APPROVED** [review]

**Comment by rosteen** [comment]
> i tested this out and it works as expected.

Thanks!

**Pull request merged**